### PR TITLE
Ignore empty registration requirements for past competitions

### PR DESCRIPTION
**Pull request by viroulep**
@Baiqiang reported on Slack that he went into some trouble when posting results for a competition, which happened because there was a validation error resulting from empty registration requirements.

I'm not sure this would be a correct fix though, as this issue with solve itself soon, when the last competitions without registration requirements will be posted.
(these are: ["Bayou Bash 2018", "Hotel Cube 2018", "Schwandorf Open 2018", "Sidoarjo Cubing Festival 2018", "Jaktim Mini Fasting 2018", "Swisscubing Cup II - 2018", "Dewata Cube Day 2018"])

@SAuroux for information, since the issue may still exist when posting results for these competition.
(Baiqiang quickfixed it by setting registration requirements to "**")

### CODE REVIEW

**Comment by jfly** [comment]
I don't really like that this changes our code to show a historical artifact (that fact that we once allowed competitions without registration requirements). What about @jonatanklosko's suggestion to do a database migration?

> Maybe it would be ok to simply set `**` for these all?

**Comment by SAuroux** [comment]
I'd be ok with that.

2018-05-25 17:34 GMT+02:00 Jeremy Fleischman <notifications@github.com>:

**Comment by viroulep** [comment]
Done :)

File: WcaOnRails/db/migrate/20180526084857_set_registration_requirements_for_upcoming_competitions.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class SetRegistrationRequirementsForUpcomingCompetitions < ActiveRecord::Migration[5.1]
+  def up
+    Competition.not_over.visible.where(registration_requirements: nil).update_all(registration_requirements: "* *")
```
Comment (jonatanklosko) [inline WcaOnRails/db/migrate/20180526084857_set_registration_requirements_for_upcoming_competitions.rb]: Actually, wouldn't it be better to do this for old comps as well? (Just to make them valid)

Comment (viroulep) [reply WcaOnRails/db/migrate/20180526084857_set_registration_requirements_for_upcoming_competitions.rb]: For all `visible` ones then?

Also rubocop is complaining that I have inserted a non breaking space in there, which is perfectly intentional, as inserting just a space is rendered as a bullet :(

Comment (jonatanklosko) [reply WcaOnRails/db/migrate/20180526084857_set_registration_requirements_for_upcoming_competitions.rb]: Yeah, all `visible` sounds good.
To workaround the error you can use other text, e.g. `"_ _"` or `"> "`.

**Comment by Baiqiang** [comment]
Sorry for a bit late. I was thinking `* *` will be rendered as `<b> </b>` but it actually is `* *` itself. Maybe some other strings that can be rendered as "blank" will be better.

**Comment by SAuroux** [comment]
Yes, indeed something that is rendered blank would be better.

By the way, Evan ran into the same issue for GreatLakesChampionship2018
tonight. I fixed this using "**" again.

2018-05-27 8:57 GMT+02:00 Baiqiang <notifications@github.com>:

**Comment by viroulep** [comment]
Sorry I've been busy this weekend, I've updated the PR!

**Comment by viroulep** [comment]
Just deployed and migrated, it looks good, so hopefully the WRT won't have issues with this again!

**Comment by SAuroux** [comment]
Thanks @viroulep !

**Comment by jfly** [comment]
Did this update future competitions as well as past and ongoing competitions? I think that might be a minor issue.

Could you also update the title of the PR to reflect the new way of doing this?

**Comment by viroulep** [comment]
It affected any visible competition with a nil registration requirements.
It should definitely be fine, as it won't change anything for these competitions (no information added or removed), and not-yet-announced competitions will have to have a non-empty registration requirements.

Do you have a specific possible issue in mind?

**Pull request merged**